Source/Hypermedia.Client.Extensions/SystemNetHttp/HttpHypermediaResolver.cs

```diff
@@ -68,27 +71,44 @@ string Quoted(string value)
                 return value;
             }
             HttpResponseMessage response;
+            bool forceRevalidate = forceResolve;
             if (this.linkHcoCache.TryGetValue(uriToResolve, out var cacheEntry))
             {
+                var isStale = cacheEntry.LocalExpirationDate == null || cacheEntry.LocalExpirationDate < DateTimeOffset.Now;
+                bool mustRevalidate = forceRevalidate
+                  || cacheEntry.CacheMode == CacheMode.AlwaysRevalidate
+                  || (isStale && cacheEntry.CacheMode == CacheMode.RevalidateStale);
+                if (!mustRevalidate)
+                {
+                    var hco = this.hypermediaReader.Read(cacheEntry.LinkResponseContent);
+                    return new ResolverResult<T>()
+                    {
+                        Success = true,
+                        ResultObject = (T)hco,
+                    };
+                }
                 var request = new HttpRequestMessage()
                 {
                     RequestUri = uriToResolve,
                     Method = HttpMethod.Get,
-                    Headers =
-                    {
-                        IfNoneMatch =
-                        {
-                            new EntityTagHeaderValue(Quoted(cacheEntry.Identifier)),
-                        }
-                    },
                 };
+                if (!string.IsNullOrEmpty(cacheEntry.Validator.ETag))
+                {
+                    request.Headers.IfNoneMatch.Add(new EntityTagHeaderValue(Quoted(cacheEntry.Validator.ETag)));
+                }
+                if (cacheEntry.Validator.LastModified != null)
+                {
+                    request.Headers.IfModifiedSince = cacheEntry.Validator.LastModified;
+                }
                 response = await this.httpClient.SendAsync(request, CancellationToken.None);
                 if (response.StatusCode == HttpStatusCode.NotModified)
                 {
+                    var hco = this.hypermediaReader.Read(cacheEntry.LinkResponseContent);
+                    // TODO: replace cache entry with possibly updated values for MaxAge
```

Comment:
Should this be resolved?

Reply:
I am not sure. It seemed to me that might be something that should be done. On the other hand if the server fails to send the cache parameters on the validation response then we should not delete the entry. Or is that the jurisdiction of the server that is able to tell us to not cache the entry anymore?
Is there even a use case, where the Server replies with "NotModified", but instructs us to not cache the value anymore? I would then expect to get a fresh response.